packages/flutter/test/material/refresh_indicator_test.dart

```diff
@@ -406,11 +386,26 @@ void main() {
       ),
     );
 
+    // test cancel
+
+    await tester.drag(find.text('A'), const Offset(0.0, expectedArmOffset - 1));
+    await tester.pump();
+    final Offset position = tester.getTopLeft(find.byType(RefreshProgressIndicator));
+    await tester.pump(const Duration(milliseconds: 1));
+    await tester.pump(const Duration(milliseconds: 1));
+    expect(tester.getTopLeft(find.byType(RefreshProgressIndicator)).dy, lessThan(position.dy));
+    expect(refreshCalled, isFalse);
+    await tester.pumpAndSettle();
+
+    // test refresh
+
     await tester.fling(find.text('A'), const Offset(0.0, 300.0), 1000.0);
     await tester.pump(const Duration(milliseconds: 100));
     expect(lastScrollOffset = controller.offset, lessThan(0.0));
     expect(refreshCalled, isFalse);
 
+    await tester.pump(const Duration(milliseconds: 100));
+    await tester.pump(const Duration(milliseconds: 100));
     await tester.pump(const Duration(milliseconds: 100));
```

Comment:
I honestly have no idea why it needs these two pumps, but when running in live binding it works even with 1 pump and without duration